scripts/ci/junit_to_github_checks.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import os
+import re
+import sys
+import textwrap
+import xml.etree.ElementTree as ET
+
+
+FILE_LINE = re.compile(r"([a-z0-9_/\\.]*):([0-9]*):")
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="convert a Pytest JUnit XML file to a checkstyle one, for reviewdog"
+    )
+    parser.add_argument("file", type=argparse.FileType("r"), default="-", nargs="?")
+    args = parser.parse_args()
+
+    tree = ET.parse(args.file)
+    root = tree.getroot()
+
+    invalid = []
+    for testcase in root.findall(".//testcase"):
+        for child in testcase:
+            if child.tag in ("error", "failure"):
```

Comment:
does this skip annotating warnings?

Reply:
Yeah it doesn't. Unfortunately, I don't think warnings are captured as a separate thing. For instance, the summary of the tags and their attributes and the counts (`[...]`) in a random Pytest file I have lying around is:

```
<testsuites> [1] 
  <testsuite> [1] errors, failures, hostname, name, skipped, tests, time, timestamp
    <testcase> [890] classname, name, time
      <skipped> [12] message, type
      <system-out> [138] 
      <failure> [18] message
      <system-err> [31] 
      <error> [4] message
``

Also, at the moment we end up with a lot of warnings from tests, so we'd have a lot of noise here.